server/src/azrael/query_resolvers.py

```diff
@@ -0,0 +1,126 @@
+
+
+from typing import Callable, Dict, Optional
+
+import pymongo
+from db import DB
+from tartiflette import Resolver
+
+database_name = 'ethereum-indexer'
+collection_name = '0x94D8f036a0fbC216Bb532D33bDF6564157Af0cD7-state'
+
+db = DB()
+
+def _remove_event_field(doc): 
+    del doc['event']
+    return doc
+
+def _cast_lent_event_fields(doc):
+    doc['lendingId'] = int(doc['lendingId'])
+    doc['lentAmount'] = int(doc['lentAmount'])
+    doc['maxRentDuration'] = int(doc['maxRentDuration'])
+    doc['paymentToken'] = int(doc['paymentToken'])
+    return doc
+
+def _cast_rented_event_fields(doc):
+    doc['lendingId'] = int(doc['lendingId'])
+    doc['rentDuration'] = int(doc['rentDuration'])
+    doc['rentedAt'] = int(doc['rentedAt'])
+    return doc
+
+def _cast_returned_event_fields(doc):
+    doc['lendingId'] = int(doc['lendingId'])
+    doc['returnedAt'] = int(doc['returnedAt'])
+    return doc
+
+def _cast_lending_stopped_event_fields(doc):
+    doc['lendingId'] = int(doc['lendingId'])
+    doc['stoppedAt'] = int(doc['stoppedAt'])
+    return doc
+
+def _cast_collateral_claimed_event_fields(doc):
+    doc['lendingId'] = int(doc['lendingId'])
+    doc['claimedAt'] = int(doc['claimedAt'])
+    return doc
+    
+def _parse_id(doc):
+    id = doc['_id'].split('_')
+    doc['tx_hash'] = id[0]
+    doc['tx_offset'] = id[1]
+
+    del doc['_id']
+    return doc
+
+_transform_lent_event = lambda doc: _remove_event_field(_cast_lent_event_fields(_parse_id(doc)))
```

Comment:
dataclass implementation will remove the need for all of this.

Reply:
implemented.